src/main/java/org/opensearch/flowframework/workflow/ReIndexStep.java

```diff
@@ -0,0 +1,175 @@
+        this.client = client;
+        this.flowFrameworkIndicesHandler = flowFrameworkIndicesHandler;
+    }
+
+    @Override
+    public PlainActionFuture<WorkflowData> execute(
+        String currentNodeId,
+        WorkflowData currentNodeInputs,
+        Map<String, WorkflowData> outputs,
+        Map<String, String> previousNodeInputs,
+        Map<String, String> params
+    ) {
+
+        PlainActionFuture<WorkflowData> reIndexFuture = PlainActionFuture.newFuture();
+
+        Set<String> requiredKeys = Set.of(SOURCE_INDEX, DESTINATION_INDEX);
+
+        Set<String> optionalKeys = Set.of(REFRESH, REQUESTS_PER_SECOND, REQUIRE_ALIAS, SLICES, MAX_DOCS);
+
+        try {
+            Map<String, Object> inputs = ParseUtils.getInputsFromPreviousSteps(
+                requiredKeys,
+                optionalKeys,
+                currentNodeInputs,
+                outputs,
+                previousNodeInputs,
+                params
+            );
+
+            String sourceIndices = (String) inputs.get(SOURCE_INDEX);
+            String destinationIndex = (String) inputs.get(DESTINATION_INDEX);
+            Boolean refresh = inputs.containsKey(REFRESH) ? Booleans.parseBoolean(inputs.get(REFRESH).toString()) : null;
+            Integer requestsPerSecond = (Integer) inputs.get(REQUESTS_PER_SECOND);
+            Boolean requireAlias = inputs.containsKey(REQUIRE_ALIAS) ? Booleans.parseBoolean(inputs.get(REQUIRE_ALIAS).toString()) : null;
+            Integer slices = (Integer) inputs.get(SLICES);
+            Integer maxDocs = (Integer) inputs.get(MAX_DOCS);
+
+            ReindexRequest reindexRequest = new ReindexRequest();
+            reindexRequest.setSourceIndices(sourceIndices);
+            reindexRequest.setDestIndex(destinationIndex);
+            if (refresh != null) {
+                reindexRequest.setRefresh(refresh);
+            }
+            if (requestsPerSecond != null) {
+                reindexRequest.setRequestsPerSecond(requestsPerSecond);
+            }
+            if (requireAlias != null) {
+                reindexRequest.setRequireAlias(requireAlias);
+            }
+            if (maxDocs != null) {
+                reindexRequest.setMaxDocs(maxDocs);
+            }
+            if (slices != null) {
+                reindexRequest.setSlices(slices);
+            }
+
+            ActionListener<BulkByScrollResponse> actionListener = new ActionListener<>() {
+
+                @Override
+                public void onResponse(BulkByScrollResponse bulkByScrollResponse) {
+                    logger.info("Reindex from source: {} to destination {}", sourceIndices, destinationIndex);
+                    try {
+                        if (bulkByScrollResponse.getBulkFailures().isEmpty() && bulkByScrollResponse.getSearchFailures().isEmpty()) {
+                            flowFrameworkIndicesHandler.updateResourceInStateIndex(
```

Comment:
Not sure if updating the state index is necessary here since it doesn't create a resource. Re-indexing happens on an existing source and destination index

Reply:
The resource created list is used in deprovisioning and will be integral for fine-grained provisioning, since this step doesnt create a resource and it doesn't have a deprovision/update equivalent I think we should avoid interacting with the state index entry here. I would consider this similar to the `SearchRequestStep` which just performs an action rather than creating a resource.